src/main/java/susan/command/AddCommand.java

```diff
@@ -0,0 +1,52 @@
+package susan.command;
+
+import susan.task.*;
+import susan.ui.Storage;
+import susan.ui.SusanException;
+import susan.ui.Ui;
+
+import java.io.IOException;
+
+public class AddCommand extends Command {
+    private String taskType;
+    private String[] commandParts;
+
+    public AddCommand(String type, String[] commandParts) {
+        this.taskType = type;
+        this.commandParts = commandParts;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws SusanException, IOException {
+        if (commandParts.length < 2) {
+            throw new SusanException("The description of a task cannot be empty.");
+        }
+
+        Task task;
+        switch (taskType) {
```

Comment:
Perhaps the switch case block could be formatted differently? (no indentation for case clauses)